### PR TITLE
Fixed Issues with Volume Calculations

### LINKED ISSUE #53044: Grenade pouch and tac vest causing absurd inventory display
### Describe the bug

Wearing the Tac Vest or Grenade Pouch causes the inventory "free volume" and "Volume (L)" values to display absurdly high values. Wearing both actually causes the Volume (L) maximum display to overflow and become negative.

This is presumably caused by #53009, in which is it likely applies to all wearable ammo-type containers.

### Steps to reproduce

1. Spawn Tac Vest and/or Grenade Pouch
2. Wear
3. Look at inventory

### Expected behavior

The effectively infinite volume of ammo containers should not be include in the calculations for inventory volume.

### Screenshots

image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 04d1bce [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Bionic Slots [cbm_slots],
    Extra Mutated Scenarios [extra_mut_scens],
    Military Profession Pack [military_professions]
]

### Additional context

_No response_

### LINKED ISSUE #53061: Certain ammo pouches have volume capacity 2000000L, wearing two makes available volume go negative.
### Describe the bug

As of experimental build 2021-11-26-0633 some ammo pouches have a volume capacity of 2000000L. Wearing two of these together results in a negative max volume, which makes picking up items impossible due to lack of space.

Ammo pouches affected, as far as I've found in the `ammo_pouch.json` file:

ammo pouch, quiver, quiver_birchbark, quiver_large, quiver_large_birchbark, tacvest.

All these seem to have the pocket data defined through ammo restriction like for example `"pocket_data": [ { "ammo_restriction": { "arrow": 60, "bolt": 60 }, "moves": 20 } ]`.

### Steps to reproduce

Equip any 2 of the following items together: ammo pouch, quiver, quiver_birchbark, quiver_large, quiver_large_birchbark, tacvest.
Open inventory, the max volume should now display negative, preventing any items from being picked up even if there is room in another pocket available (i.e. backpack or similar).

### Expected behavior

I expect ammo pouches that can hold a set number of a specific ammo type to have no effect on the max inventory volume.

### Screenshots

ammo_pouches

### Versions and configuration

Game version experimental 2021-11-26-0633 and newer tested.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Bugfixes "pockets now calculate max volumes more sensibly"

#### Purpose of change
fixes  #53061 fixes #53044

#### Describe the solution
this PR modifies how UI elements count max character volume, largest available container and item weight.
Specifically it sets up special handling rules for holsters with items in them and for pockets without a defined volume / weight.

Details:
- Reduces default volume somewhat to avoid overflows
- Adds a new function item_pocket::holster_full() which returns true if the pocket is a holster and contains something currently
- Pockets that have no volume defined (so its a huge number) no longer are candidates for largest volume pocket
- Pockets with no weight and no volume are no longer added to an item or characters totals for these things
- **UNLESS** the pocket has something in it and then the volume of the contained items is contributed to the total volume carried. So that you still see the volume of the items carried but don't display as having over your total volume.
- Pockets with a holster behave as normal containers until there is something in them.
- Holsters with something in them now are no longer considered for longest pocket and have their volume reduced on UI elements to be the size of the item in it. So if you have a large holster like a rifle holster it no longer will cause you to have a pile of unusable volume displaying as available in the UI
- A longtime pet peeve of mine, full holsters will no longer contribute to the "longest item" display

#### Describe alternatives you've considered
Pockets volume UI has always been a little weird. There is still lots more that could be done around favorites and such but this at least fixes the current problems now.

#### Testing
Wearing a holster: 
image

Holster has a gun in it (note the exact volume):
image

Grenade Pouch (has no volume and weight defined)
image

Equipped it provides no stats:
image

However when you actually put items in it volume increases to match the carried items:
image

tac vest all kitted out:
image

no gun means the total volume you can carry goes up (since you could fit any gun up to 1L):
image

removing the shells reduces max volume and current volume by the volume of the shells:
image

#### Additional context
It may be of value to incorporate some of this into the pickup and drop UI's as well at some point.

### CODE REVIEW

File: src/item_contents.cpp [hunk 1]
```diff
@@ -607,7 +608,9 @@ units::volume item_contents::max_containable_volume() const
 {
     units::volume ret = 0_ml;
     for( const item_pocket &pocket : contents ) {
-        if( !pocket.is_type( item_pocket::pocket_type::CONTAINER ) || pocket.is_ablative() ) {
+        // pockets that aren't traditional containers or don't have a default value shouldn't be counted for this
+        if( !pocket.is_type( item_pocket::pocket_type::CONTAINER ) || pocket.is_ablative() ||
+            pocket.holster_full() || pocket.volume_capacity() > 199999999_ml ) {
```
Comment (andrei8l) [inline src/item_contents.cpp:613]: You can do this without magic numbers
```suggestion
            pocket.holster_full() || pocket.volume_capacity() >= pocket_data::max_volume_capacity ) {
```
and define
```
static constexpr units::volume max_volume_capacity = 200000000_ml;
units::volume volume_capacity = max_volume_capacity;
```
in `item_pocket.h:377`
and
`constexpr units::volume pocket_data::max_volume_capacity;`
in `item_pocket.cpp:57`.
Same for the other volume and mass checks below.

Comment (bombasticSlacks) [reply src/item_contents.cpp:613]: Honestly it was getting late and I just didn't want to edit headers and compile again. I'll clean up the magic numbers tomorrow morning when I have more time.

**Comment by Hirmuolio** [comment]
To change default volume I believe you'll need to change this part too https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/item_pocket.cpp#L145

I think the value of `units::volume volume_capacity ` may be used only for pockets that are made in c++ (so migration, mod and corpse).

**Comment by bombasticSlacks** [comment]
You are right I didn't realize max volume was defined twice.

**Pull request merged**